ndcube/ndcube.py

```diff
@@ -545,16 +545,17 @@ def explode_along_axis(self, axis):
         # To store the resultant cube
         result_cubes = []
         # All slices are initially initialised as slice(None, None, None)
-        cube_slices = [slice(None, None, None)] * len(self.data.ndim)
+        cube_slices = [slice(None, None, None)] * self.data.ndim
         # Slicing the cube inside result_cube
         for i in range(self.data.shape[axis]):
             # Setting the slice value to the index so that the slices are done correctly.
             cube_slices[axis] = i
             # Set to None the metadata of sliced cubes.
-            sliced_cube = self[cube_slices]
+            item = tuple(cube_slices)
+            sliced_cube = self[item]
             sliced_cube.meta = None
             # Appending the sliced cubes in the result_cube list
-            result_cubes.append(self[cube_slices])
+            result_cubes.append(self[item])
```

Comment:
``` self[item]``` here should be ```sliced_cube```.

Reply:
Of course, I forget to modify this one when I created `item` to solve a bug.